### PR TITLE
docs: Augmenting DESIGN.md with high level arch; init container docs

### PR DESCRIPTION
**Pull request by draychev**
Continuing the work on the DESIGN.md document.
This PR adds explanation on how we patch the newly created Pods with 2 new containers.
Getting into details about how we apply iptables.

This PR also adds the following image:

osm-components-and-interactions

In follow-up PR I will continue to augment the document with explanation on how each component from the diagram above works.

Please mark with X for applicable areas.

- New Functionality      [ ]
- Documentation          [x]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests / CI System      [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution? `no`

### CODE REVIEW

File: DESIGN.md [hunk 1]
```diff
@@ -35,7 +35,31 @@ OSM ships out-of-the-box with all necessary components to deploy a complete serv
...
+[MutatingWebhookConfiguration](https://github.com/openservicemesh/osm/blob/release-v0.3/charts/osm/templates/mutatingwebhook.yaml)
+intercepts the
+[create](https://github.com/openservicemesh/osm/blob/release-v0.3/pkg/injector/webhook.go#L295)
+[pod](https://github.com/openservicemesh/osm/blob/release-v0.3/pkg/injector/webhook.go#L299)
+operations for [namespaces joined to the mesh](https://github.com/openservicemesh/osm/blob/release-v0.3/charts/osm/templates/mutatingwebhook.yaml#L19),
+and forwards these API calls to the
+[OSM control plane](https://github.com/openservicemesh/osm/blob/release-v0.3/charts/osm/templates/mutatingwebhook.yaml#L11).
+OSM control plane augments ([patches](https://github.com/openservicemesh/osm/blob/release-v0.3/pkg/injector/webhook.go#L202-L208))
+the Pod spec with 2 new containers.
+One is the [Envoy side car](https://github.com/openservicemesh/osm/blob/release-v0.3/pkg/injector/patch.go#L82-L86),
```
Comment (shashankram) [inline DESIGN.md:52]:
```suggestion
One is the [Envoy sidecar](https://github.com/openservicemesh/osm/blob/release-v0.3/pkg/injector/patch.go#L82-L86),
```

File: DESIGN.md [hunk 2]
```diff
@@ -35,7 +35,31 @@ OSM ships out-of-the-box with all necessary components to deploy a complete serv
...
+the other is an [init container](https://github.com/openservicemesh/osm/blob/release-v0.3/pkg/injector/patch.go#L61-L74).
+The init container is ephemeral. It executes the [init-iptables.sh bash script](https://github.com/openservicemesh/osm/blob/release-v0.3/init-iptables.sh)
+and terminates.
+The init container requires [NET_ADMIN Kernel capability](https://github.com/openservicemesh/osm/blob/release-v0.3/pkg/injector/init-container.go#L21-L25) for
+[iptables](https://en.wikipedia.org/wiki/Iptables) changes to be applied.
+OSM uses `iptables` to ensure that all inbound and outbound traffic flows through the Envoy sidecar.
+The [init container Docker image](https://hub.docker.com/r/openservicemesh/init)
+is passed as a string pointing to a container registry. This is passed via the `--init-container-image` CLI param to the OSM controller on startup. The default value defined in the [OSM Deployment chart](https://github.com/openservicemesh/osm/blob/release-v0.3/charts/osm/templates/osm-deployment.yaml#L33).
```
Comment (ksubrmnn) [inline DESIGN.md:60]: *The default value _is_ defined ...

File: DESIGN.md [hunk 3]
```diff
@@ -35,7 +35,31 @@ OSM ships out-of-the-box with all necessary components to deploy a complete serv
 
 
 
-## High-level architecture
+## OSM Components & Interactions
+![OSM Components & Interactions](./docs/images/osm-components-and-interactions.png)
+
+### Containers
+When a new Pod creation is initiated, OSM's
+[MutatingWebhookConfiguration](https://github.com/openservicemesh/osm/blob/release-v0.3/charts/osm/templates/mutatingwebhook.yaml)
```
Comment (michelleN) [inline DESIGN.md:43]: you can link to a file in the repo using an absolute path like: `/charts/osm/templates/mutatingwebhook.yaml`

Comment (draychev) [reply DESIGN.md:43]: @michelleN I'm not opposed to switching to abs paths - but I did want to pin the link to a version for which the document is written. I worry that one day the location of the file will change, the doc will lag behind and the link will be broken.
Not sure if we have a way to do both abs path along with `v0.3` in the path?

Comment (nojnhuh) [reply DESIGN.md:43]: > Not sure if we have a way to do both abs path along with v0.3 in the path?

I don't think that's possible. If we want to pin a version though, we should document to update it in the [release guide](https://github.com/openservicemesh/osm/blob/main/docs/release_guide.md).

**Pull request merged**